### PR TITLE
Ibm tool chain

### PR DESCRIPTION
**Pull request by ganesh-celestial**

### CODE REVIEW

**Review by DamianSzymanski — CHANGES_REQUESTED** [review]
@ganesh-celestial Please use apt, installation can be written with apt_package you will end with few lines of code instead of 140

File: cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# frozen_string_literal: true
+
+def installation_allowed?
+  %w[bionic focal xenial].include?(node['lsb']['codename']) && node['kernel']['machine'] == 'amd64'
```
Comment (DamianSzymanski) [inline cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb:4]: You choose OS in packer-templates and another condition is not needed.

File: cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    base_url = node['travis_build_environment']['ibm_advanced_tool_chain_apt_deb_url']
+    "#{base_url} #{node['lsb']['codename']} at#{node['travis_build_environment']['ibm_advanced_tool_chain_version']}"
+  end
+
+  execute "apt-key-add-ibm-advanced-tool-chain" do
+    command "wget --auth-no-challenge -qO - #{node['travis_build_environment']['ibm_advanced_tool_chain_apt_key_url']} | apt-key add -"
+    action :nothing
+  end
+
+  execute "apt-add-sources-list" do
```
Comment (DamianSzymanski) [inline cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb:18]: Resource `apt_repository` could be used here

**Review by DamianSzymanski — CHANGES_REQUESTED** [review]

File: cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb (hunk 1 above)
Comment (ganesh-celestial) [reply cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb]: Done

File: cookbooks/travis_build_environment/attributes/default.rb [hunk 3]
```diff
@@ -309,3 +309,20 @@
...
+
+default['travis_build_environment']['ibm_advanced_tool_chain_version'] = '14.0'
+default['travis_build_environment']['ibm_advanced_tool_chain_apt_key_url'] = ::File.join(
+  'https://public.dhe.ibm.com/software/server/POWER/Linux/toolchain/at/ubuntu/dists',
+  "#{node['lsb']['codename']}/6976a827.gpg.key"
+)
+default['travis_build_environment']['ibm_advanced_tool_chain_apt_deb_url'] =
+  'https://public.dhe.ibm.com/software/server/POWER/Linux/toolchain/at/ubuntu'
+
+case node['platform']
```
[the remark this answers is not in the document]
Comment (ganesh-celestial) [reply cookbooks/travis_build_environment/attributes/default.rb]: Done

File: cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb (hunk 2 above)
Comment (ganesh-celestial) [reply cookbooks/travis_build_environment/recipes/ibm_advanced_tool_chain.rb]: Done

**Review by DamianSzymanski — APPROVED** [review]

**Pull request merged**